### ISSUE TITLE
ASPNET Core Web API tests fail on Ubuntu

### ISSUE THREAD

**Issue by vkukke** [aspnet/Mvc#8654]
I have a ASPNET Core Web API with startup code configured as follows:-
```csharp
IWebHostBuilder builder = new WebHostBuilder()
                .UseConfiguration(config)
                .UseContentRoot(Directory.GetCurrentDirectory())
                .UseKestrel((context, options) =>
                {
                    IConfigurationSection kestrelConfig = context.Configuration.GetSection("kestrel");
                    options.Configure(kestrelConfig);
                    options.AddServerHeader = kestrelConfig.GetValue("addServerHeader", false);
                    options.Limits.MaxRequestBodySize = kestrelConfig.GetValue("maxRequestBodySize", 1024 * 1034); // allow 10 extra KB over the 1 MB payload
                    options.Limits.KeepAliveTimeout = TimeSpan.FromSeconds(kestrelConfig.GetValue("keepAliveTimeoutInSeconds", 300)); // default of 120 seconds
                })
```

My configuration file has
```json
"kestrel": {
    "addServerHeader": false,
    "maxRequestBodySize": 65536,
    "keepAliveTimeoutInSeconds": 300,
    "endpointDefaults": {
      "protocols": "http1"
    },
    "endpoints": {
      "httpEndpoint": {
        "url": "http://*:5889"
      }
    }
  },
}
```

Test code snippet
```csharp
[ClassInitialize]
        public static async Task SetupAsync(TestContext context)
        {
            startupHost = StartupHostBuilder.Build();
            await startupHost.StartAsync();
        }

        [ClassCleanup]
        public static async Task CleanupAsync()
        {
            await startupUpHost.StopAsync();
        }

        [TestMethod]
        public async Task TestApi()
        {
            HttpClient httpClient = new HttpClient();
           var result = await httpClient.GetAsync(requestUri);
        }
```

Expect TestApi to pass. However get System.Net.Http.HttpRequestException: Connection refused ---> System.Net.Sockets.SocketException: Connection refused.

The same code passes when run on Windows Agent.

What is missing?

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @vkukke.
@javiercn, can you please look into this? Thanks!

**Comment by Tratcher** [comment]
Can you share this as a runnable project on github?

**Comment by Tratcher** [comment]
My initial assumption is that the config isn't getting loaded correctly so the http endpoint is never added.

**Comment by vkukke** [comment]
Thanks for the reply. The tests pass in Azure Pipelines with Windows Agent. It fails on a Linux agent. I will prepare a github sample so that I can get better help to resolve this.

**Comment by vkukke** [comment]
I tried to provide  a repro environment but could not get the right logs to show up when the UTs run in Azure-Pipelines. So instead as a workaround I switched to using ASPNET Test Server

Old:
IWebHost myHost = MyHostBuilder.Build()
myHost.Start();
HttpClient client = new HttpClient()

New:
TestServer testServer = new TestServer(builder);
var httpClient = testServer.CreateClient();

and that worked without any issues. That's how I am going to get unblocked.
